third_party/gpus/sycl_configure.bzl

```diff
@@ -33,6 +36,31 @@ _GCC_HOST_COMPILER_PREFIX = "GCC_HOST_COMPILER_PREFIX"
 _CLANG_HOST_COMPILER_PATH = "CLANG_COMPILER_PATH"
 _CLANG_HOST_COMPILER_PREFIX = "CLANG_HOST_COMPILER_PATH"
 
+def _repo_root(repository_ctx, repo_name):
+    return str(repository_ctx.path(Label("@%s//:BUILD" % repo_name)).dirname)
```

Comment:
Please don't use `Label` inside the repository rule implementation. If you do that, the repository will be recreated multiple times, as many as there are `Label` initializations inside the implementation. All `Label`  parameters should become rule attributes, thus you can access them via `repository_ctx.attr` or `repository_ctx.files`.